substrate/frame/revive/src/vm/evm/instructions/host.rs

```diff
@@ -178,77 +134,103 @@ pub fn blockhash<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
 /// Loads a word from storage.
 pub fn sload<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
 	popn_top!([], index, context.interpreter);
-
-	let Some(value) = context.host.sload(context.interpreter.input.target_address(), *index) else {
-		context.interpreter.halt(InstructionResult::FatalExternalError);
-		return;
+	// NB: SLOAD loads 32 bytes from storage (i.e. U256).
+	gas!(context.interpreter, RuntimeCosts::GetStorage(32));
+	let key = Key::Fix(index.to_be_bytes());
+	let value = context.interpreter.extend.get_storage(&key);
+
+	*index = if let Some(storage_value) = value {
+		// sload always reads a word
+		let Ok::<[u8; 32], _>(bytes) = storage_value.try_into() else {
+			context.interpreter.halt(InstructionResult::FatalExternalError);
+			return
+		};
+		U256::from_be_bytes(bytes)
+	} else {
+		// the key was never written before
+		U256::ZERO
 	};
-
-	gas_legacy!(
-		context.interpreter,
-		gas::sload_cost(context.interpreter.runtime_flag.spec_id(), value.is_cold)
-	);
-	*index = value.data;
 }
 
-/// Implements the SSTORE instruction.
-///
-/// Stores a word to storage.
-pub fn sstore<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
+fn store_helper<'ext, E: Ext>(
+	context: Context<'_, 'ext, E>,
+	cost_before: RuntimeCosts,
+	set_function: fn(&mut E, &Key, Option<Vec<u8>>, bool) -> Result<WriteOutcome, DispatchError>,
+	adjust_cost: fn(new_bytes: u32, old_bytes: u32) -> RuntimeCosts,
+) {
 	require_non_staticcall!(context.interpreter);
```

Comment:
We always initialize the interpreter with
RuntimeFlags { is_static: false } so I don't think that will work is this tested?

Reply:
we should initialize it with self.ext.is_read_only(), do both
or remove this macro alltogether and do the check in line and ensure that no where we use the "is_static", as I think we might get refactor it later to use something that does not require all these panic! later on

Reply:
Yes, let's do that, remove the macro and just to be safe (even though we won't use it) initialize the interpreter with the correct is_static value